src/daft-core/src/array/ops/approx_count_distinct.rs

```diff
@@ -0,0 +1,69 @@
+use std::collections::HashSet;
+
+use arrow2::array::PrimitiveArray;
+use common_error::DaftResult;
+
+use crate::array::ops::as_arrow::AsArrow;
+use crate::array::ops::DaftApproxCountDistinctAggable;
+use crate::datatypes::UInt64Array;
+use crate::utils::identity_hash_set::IdentityBuildHasher;
+
+impl DaftApproxCountDistinctAggable for UInt64Array {
+    type Output = DaftResult<UInt64Array>;
+
+    fn approx_count_distinct(&self) -> Self::Output {
+        let mut set = HashSet::<_, IdentityBuildHasher>::default();
```

Comment:
`with_capacity(self.len())` would be more performant here!

Reply:
Updated!

Reply:
`HashSet::with_capacity` is only implemented for `HashSet<RandomState>`.

Reply:
ah okay, you can then use `HashMap<_, (), IdentityBuildHasher >` instead!

Reply:
Updated.